File: app/src/main/java/com/sevtinge/cemiuiler/module/mms/DisableAd.java
package com.sevtinge.cemiuiler.module.mms;

import static com.sevtinge.cemiuiler.module.mms.MmsDexKit.mMmsResultClassMap;

import android.content.Context;

import com.sevtinge.cemiuiler.module.base.BaseHook;

import java.util.List;
import java.util.Objects;

import io.luckypray.dexkit.descriptor.member.DexClassDescriptor;

public class DisableAd extends BaseHook {
    @Override
    public void init() {
        try {
            List<DexClassDescriptor> result = Objects.requireNonNull(mMmsResultClassMap.get("DisableAd"));
            for (DexClassDescriptor descriptor : result) {
                Class<?> enableAds = descriptor.getClassInstance(lpparam.classLoader);
                log("EnableAds class is " + enableAds);
                findAndHookMethod(enableAds, "j", new MethodHook() {
                    @Override
                    protected void before(MethodHookParam param) throws Throwable {
                        param.setResult(false);
                    }
                });
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }
        findAndHookMethod("com.miui.smsextra.ui.BottomMenu", "allowMenuMode", Context.class, new MethodHook() {
            @Override
            protected void before(MethodHookParam param) throws Throwable {
                param.setResult(false);
            }
        });
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/mms/DisableAd.kt
package com.sevtinge.cemiuiler.module.hook.mms

import android.content.Context
import com.github.kyuubiran.ezxhelper.EzXHelper
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.DexKit.addUsingStringsEquals
import com.sevtinge.cemiuiler.utils.DexKit.dexKitBridge
import com.sevtinge.cemiuiler.utils.log.XposedLogUtils

object DisableAd : BaseHook() {
    override fun init() {
        try {
            dexKitBridge.findClass {
                matcher {
                    addUsingStringsEquals("Unknown type of the message: ")
                }
            }.firstOrNull()?.getInstance(EzXHelper.classLoader)?.methodFinder()?.first {
                name == "j"
            }?.createHook {
                returnConstant(false)
            }
            /*val result: List<DexClassDescriptor> = Objects.requireNonNull<List<DexClassDescriptor>>(
                MmsDexKit.mMmsResultClassMap["DisableAd"]
            )
            for (descriptor in result) {
                val enableAds: Class<*> = descriptor.getClassInstance(lpparam.classLoader)
                log("EnableAds class is $enableAds")
                findAndHookMethod(enableAds, "j", object : MethodHook() {
                    @Throws(Throwable::class)
                    override fun before(param: MethodHookParam) {
                        param.setResult(false)
                    }
                })
            }*/
        } catch (e: Throwable) {
           XposedLogUtils.logE(TAG, e)
        }
        findAndHookMethod("com.miui.smsextra.ui.BottomMenu", "allowMenuMode",
            Context::class.java, object : MethodHook() {
                @Throws(Throwable::class)
                override fun before(param: MethodHookParam) {
                    param.setResult(false)
                }
            })
    }
}
